mcs/class/corlib/Microsoft.Win32/UnixRegistryApi.cs

```diff
@@ -690,7 +690,12 @@ void SetDirty ()
 
 		public void DirtyTimeout (object state)
 		{
-			Flush ();
+			try {
+				Flush ();
```

Comment:
We _may_ need a try/catch on the finalizer as well.  From reading the code, I actually don't think it's possible for the finalizer to run (since the object is rooted by the DirtyTimeout callback).  I don't think it would hurt to wrap `~KeyHandler` as well since we would have the same crashing issue there too if there's a race somewhere I didn't consider.

Reply:
To be clear though:  We only see failures from VS for Mac in the DirtyTimeout callstack.  I've never seen it from the finalizer, so I'm 100% good with this change.  Thanks!